pulsar-broker/src/main/java/org/apache/pulsar/client/impl/RawReaderImpl.java

```diff
@@ -213,7 +213,7 @@ void messageReceived(CommandMessage commandMessage, ByteBuf headersAndPayload, C
             }
             incomingRawMessages.add(
                     new RawMessageAndCnx(new RawMessageImpl(messageId, headersAndPayload), cnx));
-            tryCompletePending();
+            internalPinnedExecutor.execute(this::tryCompletePending);
```

Comment:
Is it better to also run lines 214-215 in the internalPinnedExecutor? So that the IO thread only has 1 time add queue operation. Currently, the IO thread will have 2 add queue operations.

Reply:
done